File: bench/micro/bus_dispatch.js
// JS equivalent of bus_dispatch.ap.
// Subject-keyed handler dispatch via Map<string, fn>. Mirrors
// Aperio's bus router lookup; NOT EventEmitter (which adds
// listener-array iteration overhead) because Aperio's bus is
// single-subscriber here.

class Aggregator {
    constructor() { this.count = 0; }
    onTick(t) { this.count = this.count + 1; }
}

const agg = new Aggregator();
const router = new Map();
router.set("bench.tick", agg.onTick.bind(agg));

const iters = 10_000;
const t0 = process.hrtime.bigint();
for (let i = 0; i < iters; i++) {
    const handler = router.get("bench.tick");
    handler({ n: i });
}
const elapsed = process.hrtime.bigint() - t0;
console.log(`iters=${iters}`);
console.log(`count=${agg.count}`);
console.log(`elapsed_ns=${elapsed}`);

File: bench/micro/bus_dispatch.py
"""Python equivalent of bus_dispatch.ap.
Subject-keyed handler dispatch via a dict — mirrors Aperio's
bus router lookup.
"""

import time


class Tick:
    __slots__ = ("n",)
    def __init__(self, n): self.n = n


class Aggregator:
    __slots__ = ("count",)
    def __init__(self): self.count = 0
    def on_tick(self, t): self.count = self.count + 1


agg = Aggregator()
router = {"bench.tick": agg.on_tick}

iters = 10_000
t0 = time.monotonic_ns()
for i in range(iters):
    handler = router["bench.tick"]
    handler(Tick(i))
elapsed = time.monotonic_ns() - t0
print(f"iters={iters}")
print(f"count={agg.count}")
print(f"elapsed_ns={elapsed}")
